search-commons/src/main/java/no/unit/nva/search2/ResourceAwsQuery.java

```diff
@@ -49,19 +74,202 @@ static Builder builder() {
     }
 
     public String doSearch(ResourceAwsClient queryClient) {
+        final var response = queryClient.doSearch(this);
         return
             MediaType.CSV_UTF_8.is(this.getMediaType())
-                ? fetchAsCsvText(queryClient)
-                : fetchAsPagedResponse(queryClient).toJsonString();
+                ? toCsvText(response)
+                : toPagedResponse(response).toJsonString();
+    }
+
+    public Stream<QueryBuilderSourceWrapper> createQueryBuilderStream(UserSettingsClient userSettingsClient) {
```

Comment:
Can you describe with comments what has been changed here and what is just copy paste from the other class, instead of using time to read it line for line too look for actual changes?

Reply:
I moved code developed in client class into Query class, where it belongs.
logic the same, renamed a couple of refereneces